recipes-kernel/linux/linux-imx.inc

```diff
@@ -38,3 +38,29 @@ KCONFIG_MODE="--alldefconfig"
 # We need to pass it as param since kernel might support more then one
 # machine, with different entry points
 KERNEL_EXTRA_ARGS += "LOADADDR=${UBOOT_ENTRYPOINT}"
+
+# A function to strip the new i.MX 32-bit dtb sub-folder nxp/imx in
+# KERNEL_DEVICETREE for older kernel builds.
+# Set KERNEL_DEVICETREE_IMX_32BIT_COMPATIBILITY_UPDATE = "1" to enable.
+KERNEL_DEVICETREE_IMX_32BIT_COMPATIBILITY_UPDATE ?= "0"
+python kernel_devicetree_imx_32bit_compatibility_update() {
+    import os.path
+    import re
+    if d.getVar('KERNEL_DEVICETREE_IMX_32BIT_COMPATIBILITY_UPDATE') != "1" or d.getVar('TUNE_ARCH') != "arm":
+        return
+    input = d.getVar('KERNEL_DEVICETREE').split()
+    output = ""
+    stripped = ""
+    for original in input:
+        if re.match("^nxp/imx", original):
```

Comment:
there are some machines supported in `meta-freescale` with dts in `nxp/mxs` directory as well

Reply:
Thanks! Will add it.

Reply:
@thochstein actually, as there are no subdirs in `arch/arm/boot/dts` in kernels < 6.6, we could just strip any subdirs in `original` ;-)

Reply:
Absolutely. That makes perfect sense.